tfjs-backend-webgpu/src/backend_webgpu.ts

```diff
@@ -170,30 +187,55 @@ export class WebGPUBackend extends KernelBackend {
     }
   }
 
+  /** Return refCount of a `TensorData`. */
+  refCount(dataId: DataId): number {
+    if (this.tensorMap.has(dataId)) {
+      const tensorData = this.tensorMap.get(dataId);
+      return tensorData.refCount;
+    }
+    return 0;
+  }
+
+  /** Increase refCount of a `TensorData`. */
+  incRef(dataId: DataId): void {
+    const tensorData = this.tensorMap.get(dataId);
+    tensorData.refCount++;
+  }
+
+  /** Decrease refCount of a `TensorData`. */
+  decRef(dataId: DataId): void {
+    if (this.tensorMap.has(dataId)) {
+      const tensorData = this.tensorMap.get(dataId);
+      tensorData.refCount--;
+    }
+  }
+
   write(values: backend_util.BackendValues, shape: number[], dtype: DataType):
       DataId {
-    const dataId = {};
+    const dataId = {id: this.nextDataId()};
```

Comment:
I'm not sure about this. Because webgl and cpu backend also added `id` in `DataId`.
Seems like `id` parameter is not necessary. `dataId={}` itself could represent a unique tensorInfo in webgpu backend, and we doesn't use `id` any where.

Reply:
I think it's convenient for debugging to know which tensor has problem. `dataId={}` is just an `Object`. I don't know the background. But it seems that it's better if `DataId` is directly a `number` not an `object`. Just keep it in this PR since we mainly resolve the memory issue.
@pyu10055 @lina128 may know more about this.